test/unit/cmd/helper/progress-buffer-overwrite.test.ts

```diff
@@ -0,0 +1,105 @@
+import {Socket} from 'socket.io-client';
+import * as sinon from 'sinon';
+import {expect} from 'chai';
+import {ProgressBufferOverwrite} from '../../../../src/helper/progress-buffer-overwrite.js';
+
+describe('progress buffer overwrite', () => {
+	let sandbox: sinon.SinonSandbox;
+	let mockedSocket: sinon.SinonStubbedInstance<Socket>;
+
+	beforeEach(() => {
+		sandbox = sinon.createSandbox({useFakeTimers: true});
+		mockedSocket = sandbox.createStubInstance(Socket) as sinon.SinonStubbedInstance<Socket>;
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	it('should send first message immideately', () => {
+		const progressBuffer = new ProgressBufferOverwrite(mockedSocket, 'test-id', 'measurement-id');
+
+		progressBuffer.pushProgress({rawOutput: 'a', hops: []});
+		progressBuffer.pushProgress({rawOutput: 'b', hops: []});
+		progressBuffer.pushProgress({rawOutput: 'c', hops: []});
+
+		expect(mockedSocket.emit.args.length).to.equal(1);
```

Comment:
🤔 I guess the purpose of this is checking the call count? If yes, better to use `mockedSocket.emit.callCount` which makes this a lot more clear.

Reply:
fixed 👍